spec/javascripts/components/cloud_network/cloud_network_form_spec.js

```diff
@@ -0,0 +1,95 @@
+describe('cloud-network-form', function() {
+  var $componentController, vm, miqService, API;
+
+  describe('when vm.recordId is not defined', function () {
+    beforeEach(module('ManageIQ'));
+    beforeEach(inject(function (_$componentController_, _API_, _miqService_, $q) {
+      $componentController = _$componentController_;
+      API = _API_;
+      miqService = _miqService_;
+
+      spyOn(miqService, 'miqAjaxButton');
+
+      var deferred = $q.defer();
+      spyOn(API, 'get').and.callFake(function() {return deferred.promise;});
+
+      var bindings = {cloudNetworkFormId: 'new'};
+      vm = $componentController('cloudNetworkForm', null, bindings);
+      vm.$onInit();
+    }));
+
+    it('sets newRecord to true', function () {
+      expect(vm.newRecord).toBe(true);
+    });
+
+    it('sets afterGet', function () {
+      expect(vm.afterGet).toBe(false);
+    });
+
+    it('adds a Cloud Network record', function () {
+      vm.cloudNetworkModel.name = 'newNetwork';
+      vm.cloudNetworkModel.ems_id = 1;
+      vm.cloudNetworkModel.id = 'new';
+      vm.cloudNetworkModel.cloud_tenant = {id: 'tenant_id'};
+      vm.cloudNetworkModel.admin_state_up = true;
+      vm.cloudNetworkModel.enabled = true;
+      vm.cloudNetworkModel.external_facing = false;
+      vm.cloudNetworkModel.provider_network_type = 'vxlan';
+      vm.cloudNetworkModel.shared = false;
+      vm.cloudNetworkModel.vlan_transparent = false;
+      vm.addClicked();
+      expect(miqService.miqAjaxButton).toHaveBeenCalledWith('create/new?button=add', vm.cloudNetworkModel, { complete: false });
+    });
+  });
+
+  describe('when vm.recordId is defined', function () {
+    beforeEach(module('ManageIQ'));
+    beforeEach(inject(function (_$componentController_, _API_, _miqService_, $q) {
+      $componentController = _$componentController_;
+      API = _API_;
+      miqService = _miqService_;
+
+      spyOn(miqService, 'miqAjaxButton');
+
+      var cloudNetworkFormResponse = {
+        id: 'id'
```

Comment:
missing comma here - hence the failing js specs ;)

Reply:
Oops, I missed that on Jasmina. Thanks